### PR TITLE
Fix issues in API Docs

### LINKED ISSUE #20024: Docerina fails to generate type name for anon/inline records
**Description:**
 Docerina fails to generate type name for anon/inline records

**Steps to reproduce:**

**Affected Versions:**
v1.0.0 at least

### PR DESCRIPTION
**Pull request by pramodya1994**
## Purpose
Fixes #20024 #22402 #21725 #22307

This include following PRs
#22405
#22404
#22400
#22310

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java [hunk 1]
```diff
@@ -426,6 +406,50 @@ public static void setPrintStream(PrintStream out) {
         BallerinaDocGenerator.out = out;
     }
 
+    /**
+     * Generate docs generator model.
+     *
+     * @param moduleDocList modules list which docs ti be generated
```
Comment (praveennadarajah) [inline misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java:412]:
```suggestion
     * @param moduleDocList modules list which docs to be generated
```

Comment (praveennadarajah) [reply misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java:412]: It should be "moduleDocList modules list **whose** docs to be generated" right?

File: misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java [hunk 2]
```diff
@@ -426,6 +406,50 @@ public static void setPrintStream(PrintStream out) {
...
+            module.summary = moduleDoc.summary;
+            module.description = moduleDoc.description;
+
+            // populate module constructs
+            sortModuleConstructs(moduleDoc.bLangPackage);
+            Generator.generateModuleConstructs(module, moduleDoc.bLangPackage);
+
+            // collect module's doc resources
+            resources.put(module.id, moduleDoc.resources);
+
```
Comment (praveennadarajah) [inline misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java:446]: I think we can remove this unnecessary line.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+    @Test(description = "Test @deprecated annotation for module-level union type definitions")
+    public void testDeprecatedUnionTypeDef() {
+        List<UnionType> unionTypes = testModule.unionTypes;
+        UnionType depUnionType = null;
+        UnionType nonDepUnionType = null;
+
+        for (UnionType type : unionTypes) {
+            String typeName = type.name;
+            if ("CountryCode".equals(typeName)) {
```
Comment (praveennadarajah) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java:84]:
```suggestion
            if (typeName.equals("CountryCode")) {
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+                depUnionType = type;
+            } else if ("NewCountryCode".equals(typeName)) {
```
Comment (praveennadarajah) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java:86]:
```suggestion
            } else if (typeName.equals("NewCountryCode")) {
```

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java (hunk 3 above)
Comment (pramodya1994) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java:84]: In the repo, we have used it in this way. That's why I used the same convention.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java (hunk 4 above)
Comment (pramodya1994) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/documentation/DeprecatedAnnotationTest.java:86]: In the repo, we have used it in this way. That's why I used the same convention.

File: misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java (hunk 2 above)
Comment (pramodya1994) [reply misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java:446]: There is no unnecessary line here.

**Pull request merged**